doc/src/manual/performance-tips.md

```diff
@@ -1394,6 +1394,116 @@ Prominent examples include [MKL.jl](https://github.com/JuliaLinearAlgebra/MKL.jl
 These are external packages, so we will not discuss them in detail here.
 Please refer to their respective documentations (especially because they have different behaviors than OpenBLAS with respect to multithreading).
 
+## Execution latency, loading and precompiling time
+
+### Reducing time to first plot etc.
+
+The first time a julia function is called it will be compiled. The [`@time`](@ref) macro family illustrates this.
+
+```
+julia> foo() = rand(2,2) * rand(2,2)
+foo (generic function with 1 method)
+
+julia> @time @eval foo();
+  0.252395 seconds (1.12 M allocations: 56.178 MiB, 2.93% gc time, 98.12% compilation time)
+
+julia> @time @eval foo();
+  0.000156 seconds (63 allocations: 2.453 KiB)
+```
+
+Note that `@time @eval ...` form is explained in the [`@time`](@ref) documentation.
+
+When developing a package, it can be helpful to cache this compilation in the package caches that are "precompiled"
+during the `Pkg` installation process or before package load. To precompile package code effectively, it's recommended
+to use [`PrecompileTools.jl`](https://julialang.github.io/PrecompileTools.jl/stable/) to run a "precompile workload" during
+precompilation time that is representative of typical package usage, which will cache the native compiled code into the
+package `pkgimage` cache, greatly reducing "time to first ..." (often referred to as TTFX) for such usage.
+
+
+### Reducing package loading time
+
+When keeping the time taken to load the package down is usually helpful.
+General good practice for package developers includes:
+
+1. Balancing the need to depending on packages for functionality vs. their load times, including stdlibs which on newer
+   julia versions may not be loaded already as stdlibs are no longer always included in the default sysimage.
```

Comment:
Maybe nitpicking but it's not "stdlibs are no longer always included in the default sysimage", but "some stdlibs are not included in the default sysimage anymore" (otherwise it sounds to me like certain stdlibs might or might not be included under certain circumstances, which isn't true I think)

Reply:
From the phrasing ("here's a consideration for you to keep in mind"), it may not be obvious what the actionable step here is. I assume you're encouraging people to think about whether you really need PkgX? I would phrase each of these as a concrete action: e.g.,
1. Reduce your dependencies to those you really need
2. Avoid use of `__init__` functions unless there is no alternative, especially... (side note: I'm pretty sure one can `@compile_workload __init__()`)
3. Where possible, fix invalidations among your dependencies and from your package code (might want a link to something that defines what invalidations are)